awx/main/models/base.py

```diff
@@ -15,7 +15,7 @@
 
 # AWX
 from awx.main.utils import encrypt_field, parse_yaml_or_json
-from awx.main.constants import CLOUD_PROVIDERS
+from awx.main.utils.plugins import CLOUD_PROVIDERS
```

Comment:
```suggestion
from awx.main.utils.plugins import discover_available_cloud_provider_plugin_names
```
and call it below.
Though, it seems that it's being used during import time which has the same problem with the execution order and that `CLOUD_INVENTORY_SOURCES` construction and use need to be refactored similarly.